### PR TITLE
Update handling of `JavaInfo.compilation_info.javac_opts`

### PR DESCRIPTION
**Pull request by hvadehra**
To optimize memory retained by Bazel, the type of `JavaInfo.compilation_info.javac_opts` is being changed from a list of tokenized strings to a depset of de-tokenized strings.

In order to use the utility method from `@rules_java` to convert this back into a list, this change also needed the following additional changes:

- Update rules_java version to 6.5.2 (the version that has the utility)
- Update bazel version(s) (the new rules_java requires a bazel version >= 6.4.0). (in any case, now that bazel 6.4.0 has been released, 6.3.0 is unsupported)

### CODE REVIEW

**Comment by jaysachs** [comment]
go/gcbrun

**Comment by jaysachs** [comment]
Presubmit is failing with:

```
Step #0 - "bazel-minversion": Repository rule http_archive defined at:
Step #0 - "bazel-minversion":   /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/bazel_tools/tools/build_defs/repo/http.bzl:372:31: in <toplevel>
Step #0 - "bazel-minversion": ERROR: /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/rules_java/toolchains/BUILD:223:14: While resolving toolchains for target @rules_java//toolchains:platformclasspath: No matching toolchains found for types @bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type.
Step #0 - "bazel-minversion": To debug, rerun with --toolchain_resolution_debug='@bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type'
Step #0 - "bazel-minversion": If platforms or toolchains are a new concept for you, we'd encourage reading https://bazel.build/concepts/platforms-intro.
Step #0 - "bazel-minversion": ERROR: Analysis of target '//buildenv/java:toolchain_jdk_17' failed; build aborted: 
Step #0 - "bazel-minversion": INFO: Elapsed time: 120.430s
Step #0 - "bazel-minversion": INFO: 0 processes.
Step #0 - "bazel-minversion": FAILED: Build did NOT complete successfully (476 packages loaded, 9629 targets configured)
Step #0 - "bazel-minversion": ERROR: Couldn't start the build. Unable to run tests
```

**Comment by hvadehra** [comment]
TLDR: It's a bit of a guess without the complete log, but if my diagnosis is correct, it should work now. 

It appears this repo too is using `github.com/bazelbuild/bazel-toolchains/pkg/rbeconfigsgen` for RBE. I had been led to believe that was only used by Bazel CI. I've added the same workaround we're currently using there. I'm not thrilled about having to do this. If y'all are averse to this, I'll see if I can cherry-pick the needed change onto rules_java 6.5.1. That would have the added benefit of not requiring all the additional, unrelated changes ( (although anyone using bazel 6 should really be on 6.4.0).

**Comment by hvadehra** [comment]
Friendly ping :)

**Comment by salguarnieri** [comment]
```
Step #0 - "bazel-minversion": ERROR: /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/rules_java/toolchains/BUILD:223:14: While resolving toolchains for target @rules_java//toolchains:platformclasspath: No matching toolchains found for types @bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type.
Step #0 - "bazel-minversion": To debug, rerun with --toolchain_resolution_debug='@bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type'
Step #0 - "bazel-minversion": If platforms or toolchains are a new concept for you, we'd encourage reading https://bazel.build/concepts/platforms-intro.
Step #0 - "bazel-minversion": ERROR: Analysis of target '//buildenv/java:toolchain_jdk_11' failed; build aborted: 
Step #0 - "bazel-minversion": INFO: Elapsed time: 128.364s
Step #0 - "bazel-minversion": INFO: 0 processes.
Step #0 - "bazel-minversion": FAILED: Build did NOT complete successfully (508 packages loaded, 4397 targets configured)
Step #0 - "bazel-minversion": ERROR: Couldn't start the build. Unable to run tests
Finished Step #0 - "bazel-minversion"
ERROR
ERROR: build step 0 "gcr.io/kythe-repo/bazelisk-builder-client" failed: step exited with non-zero status: 1
```

**Comment by hvadehra** [comment]
Alright, this is practically impossible to debug without being able to repro. I've gone ahead and made a release of rules_java based on the currently used version (6.5.1) with the necessary changes.

**Comment by jaysachs** [comment]
go/gcbrun

**Comment by hvadehra** [comment]
I'm hoping the timeout is just here on github and not in the GCP build (which would have taken longer than usual because it's the first build with a new bazel version)?

**Comment by jaysachs** [comment]
go/gcbrun

**Review by shahms — APPROVED** [review]
This seems reasonable to me, thanks!

**Comment by hvadehra** [comment]
Hi, does this need anything else, or can it be merged?

**Comment by shahms** [comment]
Just needed to wait for the CI to finish.
